src/Form/inputTypeConfig.js

```diff
@@ -32,6 +39,8 @@ const inputTypeConfig = {
         getIconBase: () => getDaxImg,
         getIconFilled: () => getDaxImg,
         shouldDecorate: (_input, {device}) => {
+            if (!canBeDecorated(_input)) return false
```

Comment:
This is added to all `shouldDecorate` functions. If you can think of a more elegant way of doing this without all the repetition, I'd be happy to change it.